### PR TITLE
Add a job to ci.yml that does `cargo install --locked --path ./zebrad/ zebrad`

### LINKED ISSUE #3449: Run `cargo install --locked --git https://github.com/ZcashFoundation/zebra --tag <new-tag> zebrad` from our README before/on release tagging
## Motivation

We want to make sure that our install instructions are currently 'green' when we update them for a new release.

### Open Questions

- This might depend on the latest release tag, should this test only run when a tag is pushed?
- Should this run every time the `README.md` file is changed?

### PR DESCRIPTION
**Pull request by dconnolly**
## Motivation

A change got into our beta.6 release where a `zebra-consensus` depended upon a test-only method in `zebra-state`, so that when a user installs zebrad via `cargo install --locked`, it fails to compile.

## Solution

We do not test this instruction as part of CI. This job addition to the `ci.yml` workflow does this, with no caching or anything, to simulate a user installing zebrad per our README instructions.

Closes #3449.
Depends-On: #4000

## Review

@teor2345 @gustavovalverde 

### Reviewer Checklist

  - [x] Catches the build error
  - [ ] Simulates the installation instructions we give to users in README.md

### CODE REVIEW

File: .github/workflows/ci.patch.yml
Comment (teor2345) [inline .github/workflows/ci.patch.yml]: Can we remove some less important info in the job name?

```suggestion
    name: Install --locked zebrad on ubuntu-latest, no cache
```

File: .github/workflows/ci.yml
Comment (teor2345) [inline .github/workflows/ci.yml]:
```suggestion
    name: Install --locked zebrad on ubuntu-latest, no cache
```

**Review by teor2345 — COMMENTED** [review]
I think this PR might close ticket #3449, because it runs the job for every PR, including release PRs.

I can also reproduce this bug using `cargo build -p zebra-utils`. So #1364 would have caught this bug as well. (But it is a larger and slower check.)

**Comment by teor2345** [comment]
Can we create a new workflow file for this job?

In future, we might want to also run it on each new tag.

Also, its run conditions are not the same as the `ci.yml` job, there are some minor differences:
```yaml
    paths:
      # code (and tests, but this job doesn't build them)
      - '**/*.rs'
      # hard-coded checkpoints (and proptest regressions, but this job doesn't run them)
      - '**/*.txt'
      # dependencies
      - '**/Cargo.toml'
      - '**/Cargo.lock'

      # changes:
      #      
      # workflow definitions
      # TODO: change to the new workflow name
      - '.github/workflows/ci.yml'

      # removals:
      #
      # no test data snapshots - this isn't a test
      # - '**/*.snap'
      # we don't run `cargo deny` in this check
      # - '**/deny.toml'
```

**Comment by teor2345** [comment]
I opened #4000 to fix the underlying bug.

File: .github/workflows/ci.yml
Comment (gustavovalverde) [inline .github/workflows/ci.yml]:
```suggestion
    name: Install zebrad from lockfile without cache on ubuntu-latest
```

File: .github/workflows/ci.patch.yml
Comment (gustavovalverde) [inline .github/workflows/ci.patch.yml]:
```suggestion
    name: Install zebrad from lockfile without cache on ubuntu-latest
```

**Review by gustavovalverde — CHANGES_REQUESTED** [review]
I'd rather not have special characters in the name (`--` `,`) as I previously had to change names because some external tools (like Mergify) had issues parsing those.

**Comment by gustavovalverde** [comment]
I do agree with @teor2345 comment of creating another workflow, but we have a pending PR to standardize workflow and jobs names. At least from my PoV we could start with this changes, and create an issue with teor's comment https://github.com/ZcashFoundation/zebra/pull/3998#issuecomment-1082608428 as a follow-up.

**Review by gustavovalverde — APPROVED** [review]
LGTM

I'll pre-approve in case there's agreement with my previous comment.

**Comment by conradoplg** [comment]
Needs rebasing/merging after #4000 merged

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
> I do agree with @teor2345 comment of creating another workflow, but we have a pending PR to standardize workflow and jobs names. At least from my PoV we could start with this changes, and create an issue with teor's comment [#3998 (comment)](https://github.com/ZcashFoundation/zebra/pull/3998#issuecomment-1082608428) as a follow-up.

Sure, I opened a new ticket #4005.

**Review by teor2345 — APPROVED** [review]
Re-approving after updating to the latest main branch

**Comment by teor2345** [comment]
It looks like this PR hung while fetching the zcash parameters, there is no log output, not even the command. This could be a GitHub actions bug.

Here is the last log:
> 2022-03-30T20:16:56.9788000Z Cache not found for input keys: macOS-sprout-and-sapling-params

https://github.com/ZcashFoundation/zebra/runs/5760640230?check_suite_focus=true#step:11:14

I'll re-run the job.

**Comment by teor2345** [comment]
@mergifyio refresh

**Pull request merged**
